Proxy bugfix

NathanSegerlind opened the pull request:
This gets in a lot of changes to the proxy model.

Commit:
LDAWrapper changes to collect() document dictionary instead of cache()

NathanSegerlind committed:
clean_suspiciousconnects

- added scaladoc
- moved data source / analysis selector into a parsed argument
- renamed parser from LDAArgumentParser to SuspiciousConnectsArgumentParser
- cosmetic cleanup

NathanSegerlind committed:
Merge branch 'clean_suspiciousconnects' into proxy_bugfix

NathanSegerlind committed:
proxy_bugfix

Corrected some problems with proxy implementation, moved model towards that proposed in Grant Babb's HTTP ml tech spec

rabarona commented on build.sbt:
```diff
@@ -3,6 +3,7 @@ name := "oni-ml"
 version := "1.1"
 
 scalaVersion := "2.10.5"
+scalacOptions += "-target:jvm-1.7"
```
What is this?

NathanSegerlind replied:
specifying a version of the jvm (1.7)

it was meant to fix a problem, but it didn't ... it should go

rabarona commented on src/main/scala/org/opennetworkinsight/LDAWrapper.scala:
```diff
@@ -68,8 +67,9 @@ object LDAWrapper {
+
     sys.process.Process(Seq(mpiCmd, "-n", mpiProcessCount, "-f", "machinefile", "./lda", "est", "2.5",
-      mpiTopicCount, "settings.txt", mpiProcessCount, modelFile, "random", localPath), new java.io.File(ldaPath)).!!
+      mpiTopicCount, "settings.txt", mpiProcessCount, modelFile, "random", localPath), new java.io.File(ldaPath)) #> (System.out) !!
```
what does #> (System.out) is needed here?

NathanSegerlind replied:
I'll clip it out

NO... 

it's there so we can get the streaming output from the MPI process running LDA instead of getting the output as a string after completion

if we want less spew, we could toggle the redirect with a conditional check, but I like seeing the MPI spew there for debuggin/sanity purposes (since so often the problem with an ML run is that MPI hangs)

rabarona commented on src/main/scala/org/opennetworkinsight/LDAWrapper.scala:
```diff
@@ -35,13 +34,13 @@ object LDAWrapper {
-    val distinctDocument = documentWordData.map(row => row(0)).distinct
-    distinctDocument.cache()
+    val distinctDocument = documentWordData.map(row => row(0)).distinct.collect
+    //distinctDocument.cache()
```
So out of the 2 options you decided .collect solution is returning better results than .cache?

NathanSegerlind replied:
set of distinct documents should fit in memory easy, so I opted for simpler solution

rabarona commented on src/main/scala/org/opennetworkinsight/ProxyPostLDA.scala:
```diff
@@ -1,60 +1,37 @@
-        }
-      }
-      df = df.select("proxy_date",
+    val dataFrame = sqlContext.parquetFile(inputPath)
```
I think dataFrame is very generic, I like the ones like scoredDF or filteredDf. Wouldn't be better something like rawDataDF?

NathanSegerlind replied:
rawDataDF it is

rabarona commented on src/main/scala/org/opennetworkinsight/ProxyPostLDA.scala:
```diff
@@ -74,48 +51,73 @@ object ProxyPostLDA {
+    logger.info("Computing conditional probability")
+    val scoredDF : DataFrame = score(sc, dataFrame, topicCount, ipToTopicMix, wordsToProbPerTopic)
+
+    val filteredDf = scoredDF.filter("score < " + threshold)
```
Would be great to keep consistency across names, even establish our names convention for RDDs and DF. In this case I think filteredDf would look better as filteredDF. 

Not a big deal.

rabarona commented:
I like you're using DataFrames for Proxy.

LGTM.

NathanSegerlind commented:
I found that using datframes really reduced the size of the code compared to the dns and flow pipelines... it's nice for adding/accessing named fields

NathanSegerlind committed:
proxy_bugfix

- removed JVM specification from build file

- variable renamings

- eliminated ignored unit tests
